replica/dirty_memory_manager.hh

```diff
@@ -564,12 +567,7 @@ template <typename Func>
 requires (!is_future<std::invoke_result_t<Func>>::value)
 futurize_t<std::invoke_result_t<Func>>
 region_group::run_when_memory_available(Func&& func, db::timeout_clock::time_point timeout) {
-    bool blocked = 
-        !_blocked_requests.empty()
-        || under_unspooled_pressure()
-        || _under_real_pressure;
-
-    if (!blocked) {
+    if (!_blocked_requests.empty() || !execution_permitted()) {
         return futurize_invoke(func);
```

Comment:
You lost (or gained) a negation. It even reads wrong: if not execution permitted, invoke func.

Reply:
And, || -> &&.

Reply:
True :( Gained one more negation back

Reply:
Praise De Morgan!